api/endpoints.go

```diff
@@ -198,6 +198,17 @@ func handleSession(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		existingSave, err := savedata.GetSession(uuid, slot)
+		if err != nil && !errors.Is(err, sql.ErrNoRows) {
+			httpError(w, r, fmt.Errorf("failed to retrieve session save data: %s", err), http.StatusInternalServerError)
+			return
+		} else {
+			if existingSave.Seed == session.Seed && existingSave.WaveIndex > session.WaveIndex {
+				httpError(w, r, fmt.Errorf("session out of date: existing wave index is greater"), http.StatusBadRequest)
+				return
+			}
+		}
```

Comment:
You dont return an errror here for the else of that if. So you potentially lose errors here

Reply:
Everything outside of the checks is expected behavior.